management-api-agent-common/src/main/java/io/k8ssandra/metrics/config/ConfigReader.java

```diff
@@ -0,0 +1,34 @@
+package io.k8ssandra.metrics.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import java.io.File;
+import java.io.IOException;
+
+public class ConfigReader {
+    public static final String CONFIG_PATH_PROPERTY = "collector-config-path";
+    public static final String CONFIG_PATH_DEFAULT = "/opt/management-api/configs/metric-collector.yaml";
+
+    public static Configuration readConfig() {
+        // Check env variable if there's any changes to the config path
+        String configPath = System.getProperty(CONFIG_PATH_PROPERTY);
+        if(configPath == null) {
+            configPath = CONFIG_PATH_DEFAULT;
+        }
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        File configFile = new File(configPath);
+
+        // FileNotFoundException should be thrown if override is used
+        if(configFile.exists() || !configPath.equals(CONFIG_PATH_DEFAULT)) {
```

Comment:
This condition isn't easy to understand.
Why don't we read the config if it's under the default path? Is it empty by default? Can't we have the case where we have some custom config loaded under the default path?

Reply:
The config does not exists by default, so in that case we don't want to throw Exception even if it can't be read. If the path has been modified, I'd expect the file to exists.